File: mss/frontend/src/types/review.ts
import type { UserBrief } from "./user";

export interface Review {
  id: string;
  game_id: string;
  user: UserBrief;
  version_reviewed?: string | null;
  score: number;
  body: string;
  helpful_count: number;
  not_helpful_count: number;
  is_edited: boolean;
  created_at: string;
  updated_at: string;
}

export interface ReviewCreate {
  score: number;
  body: string;
}

File: mss/backend/app/schemas/review.py
from __future__ import annotations

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field

from app.schemas.user import UserBriefResponse


class ReviewCreate(BaseModel):
    score: int = Field(ge=1, le=10)
    body: str = Field(min_length=1)


class ReviewUpdate(BaseModel):
    score: int | None = Field(default=None, ge=1, le=10)
    body: str | None = Field(default=None, min_length=1)


class ReviewResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    game_id: UUID
    user: UserBriefResponse
    version_reviewed: str | None = None
    score: int
    body: str
    helpful_count: int = 0
    not_helpful_count: int = 0
    is_edited: bool = False
    created_at: datetime
    updated_at: datetime


class ReviewVoteRequest(BaseModel):
    is_helpful: bool
